torch/csrc/api/include/torch/nn/modules/sequential.h

```diff
@@ -302,7 +320,12 @@ class SequentialImpl : public Cloneable<SequentialImpl> {
   /// pack has only one type, in which case the template would be preferred,
   /// even if the other `push_back` functions are better fits (e.g. `unique_ptr`
   /// -> `shared_ptr` overload).
-  template <typename First, typename Second, typename... Rest>
+  /// NOTE: We explicitly avoid matching this template with `push_back(std::string("name"), module)`
+  /// or `push_back("name", module)`, since they should be handled by their respective
+  /// `push_back` functions.
+  template <typename First, typename Second, typename... Rest,
+    typename = torch::disable_if_t<std::is_same<First, std::string>::value ||
```

Comment:
Why do we need the `const char*` guard? I think we're only using `std::string` in here?

Reply:
If we don't add the guard for `const char*` here, a call such as `sequential->push_back("shared_m1", M(1))` will be template-matched to this `push_back(First&& first, Second&& second, Rest&&... rest)` method, which is not what we want (we want it to match to `push_back(std::string name, M&& module)` instead.